### DISCUSSION TITLE
trackEnd event

### DISCUSSION THREAD

**Discussion by aka-pal** [Androz2091/discord-player#621] [Proposals]
**Is your feature request related to a problem? Please describe.**

It's not a discord-player problem, but mine. I'm using reaction controls , so whenever a track ends I've to delete the old play message. But there is not track end event, i can make detect when a track ends. Please please add a trackEnd event
**Describe the solution you'd like**

**Describe  #alternatives you've considered**

Add a trackEnd event, whenever a track ends or skipped, jumped it'll be emitted
**Additional context**

**Comment by Positive2022** [comment]
bro you can just use `setTimeout` in track start event and use `track.durationMS` as the time in setTimeout function :D
code would look like this:
```js
var msg = await message.channel.send('started playing song')
setTimeout( async () => {
if (msg && !msg.deleted) return msg.delete()
}, track.durationMS)
```

**Comment by aka-pal** [comment]
What if someone pause for some time
And if someone skiped the track ?

**Comment by Positive2022** [comment]
then i would just do `client.player.emit('trackEnd', async (client, message, msg_to_delete) => {
// and check if messages exists and delete it.
})
YES. You can emit your own events if you want.
But i would still support your point of adding trackend event :)

**Comment by sayuriu** [comment]
when a track ends [either by error or skipping (even in pause)], either `queueEnd` or `trackStart` or `error` will be emitted
you can attach the current player message to `Queue<metadata>` so that can be handled
that makes `trackEnd` is kinda not needed
- this is just an opinion

**Comment by aka-pal** [comment]
But when a track start for the first time , trackStart event also emitted

**Comment by sayuriu** [comment]
`trackStart` is emitted when whatever track starts playing
you can attach `message` object to the `Queue` object as cache
it will not be reset when another `trackStart` emits, unless if you destroy the queue
